ddtrace/contrib/wsgi/wsgi.py

```diff
@@ -151,6 +163,12 @@ def blocked_view():
             result = core.dispatch_with_results(
                 "wsgi.request.complete", (ctx, closing_iterable, self.app_is_iterator)
             ).traced_iterable
+
+            if stop_iteration_exception and result.value:
+                # Close the request and app spans
+                result.value._finish_spans()
+                core.dispatch("wsgi.app.success", (ctx, closing_iterable))
```

Comment:
```suggestion
            if stop_iteration_exception:
                if result.value:
                    # Close the request and app spans
                    result.value._finish_spans()
                    core.dispatch("wsgi.app.success", (ctx, closing_iterable))
```


Are there cases when we have a stop iteration exception but not a result value?


Should we always be raising the stop iteration exception even if we don't have a result value?

Reply:
Whoops! Your code is more correct, the intent was to be defensive and I messed it up - accepting your fix